app/uk/gov/hmrc/agentaccesscontrol/connectors/desapi/DesAgentClientApiConnector.scala

```diff
@@ -83,9 +89,13 @@ class DesAgentClientApiConnector @Inject()(
 
   def getAgentRecord(agentId: TaxIdentifier)(
       implicit hc: HeaderCarrier,
-      ec: ExecutionContext): Future[AgentRecord] = {
+      ec: ExecutionContext): Future[Either[String, AgentRecord]] = {
     getWithDesHeaders[AgentRecord]("GetAgentRecord",
                                    new URL(getAgentRecordUrl(agentId)))
+      .map(a => Right(a))
+      .recover {
+        case e: Upstream4xxResponse => Left(e.message)
```

Comment:
do we want the application to blow up in case of other errors ?

Reply:
Do you think we should only explicitly handle like this if the e.message contains "AGENT_TERMINATED"? Every other 4xx would then get transformed into a 500 response from agent-access-control

Reply:
yeah , check for e.message contains "AGENT_TERMINATED" and return accordingly